### ISSUE TITLE
Fix GraphConcatSpec

### ISSUE THREAD

**Issue by drewhk** [akka/akka#16938]
There are two test cases currently marked as pending because they relied on materialization order of the old materializer. These tests should be revived and adapted to the new changes.

**Comment by bantonsson** [comment]
I don't see the issue with the tests, since they test the opposite wiring of each other and should cover both materialization cases, unless our materialization order is random.

**Comment by drewhk** [comment]
Our materialization order is currently non-deterministic (not necessarily random)

**Comment by bantonsson** [comment]
Let me rephrase that. If I have a graph with exactly the same shape, do you mean that two different materializations can have different order of materialization?

**Comment by bantonsson** [comment]
Even more narrow. If I materialize the same graph twice, can the order be different?

**Comment by drewhk** [comment]
Probably not on the same JVM run, but it depends on HashSet ordering.

**Comment by bantonsson** [comment]
Check, then my comment is broken. I'll have to read up on the materialization code.

**Comment by drewhk** [comment]
Enough to look at MaterializerSession it does the traversal.

**Comment by bantonsson** [comment]
I just took a look at this, and it is deterministic since the Set only contains 3 submodules, and becomes an optimized Set3, that traverses in insertion order :wink:

I added a comment to the tests.
